docs/adr/0002-use-local-datastore.md

```diff
@@ -0,0 +1,58 @@
+# 2: Use local datastore for publisher workflow
+
+Date: 2018-07-20
+
+## Context
+We're building a new publishing architecture and MVP workflow in private beta. We have taken the decision to build this in a new application and therefore we need to decide how we use the publishing-api.
+
+One of the goals of the migration effort from 2016 was to for all publishing applications to be clients of the publishing-api and this API would means that these applications would not need local datastores. This is written up in the [GOV.UK Platform Architecture Goals 2017-18+][goals]
```

Comment:
> not need local datastores

I don't think this was ever a serious aim. The Publishing API is just for content, not everything an application might need to store.

I've always interpreted this as a unclear way of saying "only store content in the Publishing API". That's my reading of the referenced goals document as well, as it says "all content stored within the Publishing API, not relying on their own local datastores".

Reply:
Maybe I misunderstood the document, but I was under the impression that it was the aim to avoid using a local datastore. Which is why specialist publisher is built the way it is.

Reply:
> Maybe I misunderstood the document, but I was under the impression that it was the aim to avoid using a local datastore. Which is why specialist publisher is built the way it is.

In my mind, this is only for content. Specialist Publisher has a local datastore, it uses it for storing user information. We should try and clarify this, do you see it as applying to any and all data?

Reply:
I see it as being applied to all data, see here for an [expansion](https://github.com/alphagov/content-publisher/pull/25#discussion_r204361821) . Although I would say that user data is a special case as that's required by OmniAuth.